src/bindings/python/docs/code_examples.md

```diff
@@ -0,0 +1,413 @@
+```python
+from openvino.helpers.custom_module.custom_helpers import top1_index
+```
+
+Follow it with correct import in `helpers/__init__.py`:
+```python
+from openvino.helpers.custom_module import top1_index
+```
+**Do not forget to include a license on the top of each file!** For demonstrational puroposes (and to keep this markdown file short) it is not included in above snippets.
+
+To see changes taking effect, rebuild project (Cmake's install step should be good enough) and run your solution:
+```python
+import openvino.helpers as ov_helpers
+
+ov_helpers.top1_index([0.7, 2.99, 3.0, -1.0])
+>>> 2
+```
+
+Following this method developers can add new modules and adjust existing ones including structure, naming and imported functions.
+<!-- !!!!!!!!Mention about adding it to setup.py as well!!!!!!!! -->
+
+<!-- Pure pybind11 solution describes C++ based approach -->
+
+#### Pure pybind11 solution
+Second approach to extend OpenVINO:tm: codebase is utilizing *pybind11* library. It allows to write C++ based code, thus creating so-called Python bindings.
+
+**The example in this section covers the scenario of adding new features to newly created submodule. Extending existing codebase can be done in similar fashion by working on already implemented classes and modules.**
+
+###### Before start: What is pybind11?
+<!-- Do we keep it here? -->
+Link to offical documentation: https://pybind11.readthedocs.io/en/stable/
+Link to project repository: https://github.com/pybind/pybind11
+
+##### Adding new (sub)module
+Adding new module could be done only by using *pybind11* built-in capabilities.
+
+Navigate to main project file responsible for creation of whole package, let's call it "registering-point":
+
+    openvino/src/bindings/python/src/pyopenvino/pyopenvino.cpp
+
+Add new submodule by writing:
+```cpp
+py::module mymodule = m.def_submodule("mymodule", "My first feature - openvino.runtime.mymodule");
+```
+This is a shorthand way of adding new submodules which can later be used to extend package. The mysterious `m` is actaully the main OpenVINO:tm: module called `pyopenvino` -- it is registered with `PYBIND11_MODULE(pyopenvino, m)` at the top of the "registering-point" file. Later imports from it are done by calling upon `openvino.pyopenvino` package.
+
+Please keep in mind, in most of real-life scenarios, modules and classes are registered in different files. General idea is to create helper function that will hold all of the registered modules, classes and functions. This function needs to be exposed within a separate header file and included in "registering-point". Project's common guideline suggest to use names in the following convention `reg[class|module]_[domain]_[name_of_the_class|module]`. Where optional `[domain]` generally points to parts of the API like graph, frontend or stay empty in case of core runtime. Examples can be found in "registering-point" file `openvino/src/bindings/python/src/pyopenvino/pyopenvino.cpp`.
```

Comment:
What does the sign " | " mean? Is it an alternative? In my opinion splitting `reg[class|module]_[domain]_[name_of_the_class|module]` to the `reg[class]_[domain]_[name_of_the_class] or reg[module]_[domain]_[name_of_the_module]` will be more readable.

Reply:
Good idea!